spec/lib/diaspora/federated/generator.rb

```diff
@@ -0,0 +1,16 @@
+require "spec_helper"
+
+describe "comment on reshare" do
+  before do
+    @status_message = FactoryGirl.build(:status_message, author: alice.person, public: true)
+    @reshare = FactoryGirl.build(:reshare, root: @status_message, author: bob.person)
+    @comment = FactoryGirl.build(:comment, post: @reshare, text: "reshare comment")
+  end
+
+  it "delivers comment also to root author" do
+    opts = {"additional_subscribers" => [alice]}
```

Comment:
`alice` is a `User`, yet you add a `Person`, no idea how this ever passed, but it doesn't seem to test your addition.